Sources/StreamChat/Database/DTOs/PollVoteDTO.swift

```diff
@@ -0,0 +1,226 @@
+    @NSManaged var isAnswer: Bool
+    @NSManaged var answerText: String?
+    @NSManaged var option: PollOptionDTO?
+    @NSManaged var poll: PollDTO?
+    @NSManaged var user: UserDTO?
+    @NSManaged var queries: [PollVoteListQueryDTO]?
+    
+    static func loadOrCreate(
+        voteId: String,
+        poll: PollDTO,
+        option: PollOptionDTO?,
+        user: UserDTO?,
+        context: NSManagedObjectContext,
+        cache: PreWarmedCache?
+    ) -> PollVoteDTO {
+        if let cachedObject = cache?.model(for: voteId, context: context, type: PollVoteDTO.self) {
+            return cachedObject
+        }
+        
+        let request = fetchRequest(for: voteId, pollId: poll.id)
+        if let existing = load(by: request, context: context).first {
+            return existing
+        }
+
+        let new = NSEntityDescription.insertNewObject(into: context, for: request)
+        new.id = voteId
+        new.option = option
+        new.poll = poll
+        new.user = user
+        return new
+    }
+    
+    static func load(voteId: String, pollId: String, context: NSManagedObjectContext) -> PollVoteDTO? {
+        let request = fetchRequest(for: voteId, pollId: pollId)
+        return load(by: request, context: context).first
+    }
+    
+    static func fetchRequest(for voteId: String, pollId: String) -> NSFetchRequest<PollVoteDTO> {
+        let request = NSFetchRequest<PollVoteDTO>(entityName: PollVoteDTO.entityName)
+        request.predicate = NSPredicate(format: "id == %@ && pollId == %@", voteId, pollId)
+        return request
+    }
+}
+
+extension PollVoteDTO {
+    func asModel() throws -> PollVote {
+        try PollVote(
+            id: id,
+            createdAt: createdAt.bridgeDate,
+            updatedAt: updatedAt.bridgeDate,
+            pollId: pollId,
+            optionId: optionId,
+            isAnswer: isAnswer,
+            answerText: answerText,
+            user: user?.asModel()
+        )
+    }
+}
+
+extension NSManagedObjectContext {
+    @discardableResult
+    func savePollVotes(payload: PollVoteListResponse, query: PollVoteListQuery?, cache: PreWarmedCache?) -> [PollVoteDTO] {
+        let isFirstPage = query?.pagination.offset == 0
+        if let filterHash = query?.queryHash, isFirstPage {
+            let queryDTO = PollVoteListQueryDTO.load(filterHash: filterHash, context: self)
+            queryDTO?.votes = []
+        }
+
+        return payload.votes.compactMapLoggingError {
+            if let payload = $0 {
+                return try savePollVote(payload: payload, query: query, cache: cache)
+            } else {
+                return nil
+            }
+        }
+    }
+    
+    @discardableResult
+    func savePollVote(
+        payload: PollVotePayload,
+        query: PollVoteListQuery?,
+        cache: PreWarmedCache?
+    ) throws -> PollVoteDTO {
+        guard let poll = try poll(id: payload.pollId) else {
+            throw ClientError.Unexpected()
```

Comment:
`MessageDoesNotExist` but for polls would be nice since then it prints out the id at least, e.g.`PollDoesNotExist(pollId:)`

Reply:
knew I forgot sth 😅 Yeah, new errors need to be introduced.